src/preact/context.js

```diff
@@ -93,29 +93,24 @@ export function useAmpContext() {
  * @param {boolean} unloadOnPause
  * @return {boolean}
  */
-export function useLoad(loadingProp, unloadOnPause = false) {
-  const loadingRef = useRef(false);
+export function useLoading(loadingProp, unloadOnPause = false) {
+  const {loading: loadingContext, playable} = useAmpContext();
+  const loading = loadingReducer(loadingProp, loadingContext);
 
-  const {loading: loadingContext, renderable, playable} = useAmpContext();
+  const shouldUnloadOnPauseRef = useRef(false);
+  const triggerUnloadOnPauseRef = useRef(false);
 
-  const loading = loadingReducer(loadingProp, loadingContext);
+  // Force unload only when the playable goes from true -> false.
+  const shouldUnloadOnPause = !playable && unloadOnPause;
+  if (shouldUnloadOnPause != shouldUnloadOnPauseRef.current) {
+    shouldUnloadOnPauseRef.current = shouldUnloadOnPause;
+    triggerUnloadOnPauseRef.current = shouldUnloadOnPause;
+  }
+
+  // Reset triggerUnloadOnPause when loading becomes eager.
+  if (loading == Loading.EAGER) {
+    triggerUnloadOnPauseRef.current = false;
+  }
 
-  // Compute whether the element should be loading at this time.
-  const load =
-    // Explicit instruction to unload.
-    loading == Loading.UNLOAD
-      ? false
-      : // Must be unloaded to pause.
-      unloadOnPause && !playable
-      ? false
-      : // Explicit instruction to load.
-      loading == Loading.EAGER
-      ? true
-      : // Auto: allowed to load.
-      loading == Loading.AUTO && renderable
-      ? true
-      : // Lazy: can continue loading if already started, but do not start it.
-        loadingRef.current || false;
-  loadingRef.current = load;
-  return load;
+  return triggerUnloadOnPauseRef.current ? Loading.UNLOAD : loading;
```

Comment:
Do we need a `triggerUnloadOnPauseRef` or `shouldUnloadOnPauseRef`? I think the entire function is reducible to:

```js
  const {loading: loadingContext, playable} = useAmpContext();
  const loading = loadingReducer(loadingProp, loadingContext);
  return loading == Loading.EAGER || playable || !unloadOnPause ? loading : Loading.UNLOAD;
```

Reply:
This has been simplified dramatically. It's now even simpler than your suggestion. PTAL.